espnet2/enh/espnet_model.py

```diff
@@ -161,6 +189,28 @@ def forward(
             dereverb_speech_ref = dereverb_speech_ref[..., : speech_lengths.max()]
             dereverb_speech_ref = dereverb_speech_ref.unbind(dim=1)
 
+        # sampling frequency information about the batch
+        if "utt2fs" in kwargs:
+            # All samples must have the same sampling rate
+            fs = kwargs["utt2fs"][0].item()
+            assert all([fs == kwargs["utt2fs"][0].item() for fs in kwargs["utt2fs"]])
+
+            # Adaptively adjust the STFT encoder/decoder window/hop sizes
+            if isinstance(self.separator, USESSeparator):
+                if isinstance(self.encoder, STFTEncoder):
+                    self.encoder.reconfig_for_fs(fs)
```

Comment:
I suggest removing this call and passing `fs` to the encoder's `forward` function directly.

Reply:
Good idea!

Reply:
done